provision.sh

```diff
@@ -1,31 +1,138 @@
-echo "Installing python pip"
-echo "====================="
-wget -nv -O - https://bootstrap.pypa.io/get-pip.py | python
-echo "Installing Splash"
-echo "================="
-/vagrant/docker/splash.sh
-echo "Installing portia dependencies"
-echo "=============================="
-cd /vagrant
-pip install -q -r slybot/requirements.txt
-pip install -q -r slyd/requirements.txt
-pip install -q -e /vagrant/slybot
-pip install -q -e /vagrant/slyd
-echo "Installing slyd as a Upstart service"
-echo "===================================="
-cp /vagrant/slyd.conf /etc/init
-echo "Starting slyd service"
-echo "====================="
-start slyd
+set -e
+
+if [ "x$APP_ROOT" = x ]
+then
+    for dir in "$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )" /app /vagrant
+    do
+        if [ -d "$dir" ] && [ -d "$dir/slyd" ]
+        then
+            APP_ROOT="$dir"
+            break
+        fi
+    done
+fi
+
+if [ "x$APP_ROOT" = x ]
+then
+    echo "Could not determine app directory"
+    exit 1
+fi
+
+echo "APP_ROOT=$APP_ROOT"
+
+usage() {
+    cat <<EOF
+Portia provisioner.
+
+Usage: $0 COMMAND [ COMMAND ... ]
+
+Available commands:
+usage -- print this message
+install_deps -- install system-level dependencies
+install_splash -- install splash
+install_python_deps -- install python-level dependencies
+configure_initctl -- installs initctl configuration and starts slyd
+configure_nginx -- installs nginx configuration
+cleanup -- remove unnecessary files. DON'T RUN UNLESS IT'S INSIDE AN IMAGE AND YOU KNOW WHAT YOU ARE DOING
+
+EOF
+}
+
+export SPLASH_PYTHON_VERSION=${SPLASH_PYTHON_VERSION:-"2"}
+
+activate_venv () {
+    if [ -e ${VIRTUAL_ENV}/bin/activate ]; then
+        echo "Activating virtual env..."
+        source ${VIRTUAL_ENV}/bin/activate
+    fi
+}
+
+install_deps(){
+    echo deb http://nginx.org/packages/ubuntu/ trusty nginx > /etc/apt/sources.list.d/nginx.list
+    apt-key adv --keyserver keyserver.ubuntu.com --recv-keys ABF5BD827BD9BF62
+    apt-get update
+    apt-get -y install \
+            curl \
+            libxml2-dev \
+            libxslt-dev \
+            libgl1-mesa-dev-lts-utopic \
+            libgl1-mesa-glx-lts-utopic \
+            libglapi-mesa-lts-utopic \
+            libgl1-mesa-dri-lts-utopic \
+            nginx python-dev \
+            python-mysql.connector \
+            python-numpy \
+            python-openssl \
+            python-pip \
+            python-software-properties
+}
+
+install_python_deps(){
+    activate_venv
+    pip install http://cdn.mysql.com/Downloads/Connector-Python/mysql-connector-python-1.2.3.zip#md5=6d42998cfec6e85b902d4ffa5a35ce86
+    pip install -r "$APP_ROOT/slyd/requirements.txt"
+    pip install -r "$APP_ROOT/slybot/requirements.txt"
+    pip install -e "$APP_ROOT/slyd"
+    pip install -e "$APP_ROOT/slybot"
+}
+
+install_splash(){
+    cd /tmp
+    curl -L -o splash.tar.gz 'https://github.com/scrapinghub/splash/archive/master.tar.gz'
```

Comment:
Yes, once splash 2.0 is out. We should think about how to deploy this, if we want to release for the new UI only or make an old UI release with Qt5. Either way should cause no problem, just merging this pull request into the new UI branch should work since this doesn't change anything in the UI.